tests/unit/test_endpoint.py

```diff
@@ -99,12 +100,6 @@ def test_make_request_with_no_auth(self):
         self.assertNotIn('Authorization', prepared_request.headers)
 
     def test_make_request_no_signature_version(self):
-        self.endpoint = Endpoint(
```

Comment:
True. Those lines are duplicated in the base class. In my first prototype, I copy-and-paste them into my new test case, and then noticed then-3-copies of such snippets in subclass. Bad smell.